debezium-core/src/main/java/io/debezium/relational/Tables.java

```diff
@@ -58,6 +58,34 @@ static TableFilter fromPredicate(Predicate<TableId> predicate) {
         static TableFilter includeAll() {
             return t -> true;
         }
+
+        /**
+         * Creates a {@link TableFilter} that caches the results of the given filter.
+         *
+         * @param filter the filter to wrap; may not be null
+         * @return a caching filter
+         */
+        static TableFilter cached(TableFilter filter) {
+            return new CachingTableFilter(filter);
+        }
+    }
+
+    /**
+     * A {@link TableFilter} that caches the results of another filter.
+     */
+    private static final class CachingTableFilter implements TableFilter {
+
+        private final TableFilter delegate;
+        private final ConcurrentMap<TableId, Boolean> cache = new ConcurrentHashMap<>();
+
+        CachingTableFilter(TableFilter delegate) {
+            this.delegate = Objects.requireNonNull(delegate);
+        }
+
+        @Override
+        public boolean isIncluded(TableId tableId) {
+            return cache.computeIfAbsent(tableId, delegate::isIncluded);
```

Comment:
in case of config update, say a filter is removed, how are we handling it? Don't we need to explicitly remove them ?

Reply:
I am thinking that on config update, Task would be restarted so all in memory objects would be recreated.

Reply:
yeah, as long during connector close, all the refrences are cleaned up, GC would reclaim this.